activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter-hal-rest-api/src/main/java/org/activiti/controllers/ProcessDefinitionController.java

```diff
@@ -72,4 +77,69 @@ public ProcessDefinitionResource getProcessDefinition(@PathVariable String id) {
         }
         return resourceAssembler.toResource(processDefinitionConverter.from(processDefinition));
     }
+    
+    @RequestMapping(value = "/{id}/meta", method = RequestMethod.GET)
+    public ProcessDefinitionResource getProcessDefinitionMetadata(@PathVariable String id) {
+    	org.activiti.engine.repository.ProcessDefinition processDefinition = repositoryService.createProcessDefinitionQuery().processDefinitionId(id).singleResult();
+        if (processDefinition == null) {
+            throw new ActivitiException("Unable to find process definition for the given id:'" + id + "'");
+        }
+        Process process = repositoryService.getBpmnModel(id).getMainProcess();
+        List<String []> variables = new ArrayList<String[]>();
+        if(!process.getExtensionElements().isEmpty())
+        {
+        	Iterator<List<ExtensionElement>> it = process.getExtensionElements().values().iterator();
+        	while(it.hasNext())
+        	{
+        		List<ExtensionElement> extensionElementList = it.next();
+        		Iterator<ExtensionElement> it2 = extensionElementList.iterator();
+        		while(it2.hasNext())
+        		{
+        			ExtensionElement ee = it2.next();
+        			String[] variable = new String[2];
+        			variable[0] = ee.getAttributeValue(ee.getNamespace(), "variableName");
+        			variable[1] = ee.getAttributeValue(ee.getNamespace(), "variableType");
```

Comment:
Same comment from my side, I would create a class to represent that instead of an array, is there any particular reason why you choose an array?